src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -968,7 +968,7 @@ bool Aura::ModStackAmount(int32 num, AuraRemoveMode removeMode, bool periodicRes
         return true;
     }
 
-    bool refresh = stackAmount >= GetStackAmount();
+    bool refresh = stackAmount >= GetStackAmount() && (m_spellInfo->StackAmount || !m_spellInfo->HasAttribute(SPELL_ATTR1_DONT_REFRESH_DURATION_ON_RECAST));
```

Comment:
this change is not "just redifining names" so it needs to be properly tested

Reply:
That attribute just does so that spells that have it, do not refresh duration by the code below.

So nothing is changed really, just making use of the attribute SPELL_ATTR1_**DONT_REFRESH_DURATION_ON_RECAST**

Reply:
How to test here

Reply:
I guess we need to test the "spells that have it"... @Kitzunu can you provide just a few examples so we can have testers on them?

Reply:
If I can get spellwork to search like a normal human being then sure